packages/model-viewer/src/features/annotation.ts

```diff
@@ -132,9 +132,12 @@ export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
      * model-viewer element. The position and normal are returned as strings in
      * the format suitable for putting in a hotspot's data-position and
      * data-normal attributes. If the mesh is not hit, the result is null.
+     * 
+     * The third return value is the current hit material intersected at the pixel coordinates.
+     * This is usefull to apply materials changes on mouse events (hover/click).
      */
     positionAndNormalFromPoint(pixelX: number, pixelY: number):
-        {position: Vector3D, normal: Vector3D}|null {
+      {position: Vector3D, normal: Vector3D, material?: MeshStandardMaterial }|null {
```

Comment:
`MeshStandardMaterial` is still an internal three.js type that we do not want to expose publicly in MV. You should instead return our `Material` wrapper object: https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/features/scene-graph/api.ts#L42

Reply:
Maybe now I have understood what you mean... sorry. I have just updated and pushed the code.

Is there some documentation about how is structured the code? Sort of diagrams, schemas, technical descriptions or best practices to adopt for contributing? I didn't find something like that in the repo, thanks!

Reply:
Hi, I'm currently It looks like this PR will be blocked until we have more of the SceneGraph support working which is in progress, I currently have PR #2816 that is going to let us address individual Three.js mesh primitives as wrapped Model-Viewer primitives, once this is in place we can create conversion helper from the Three Mesh/Primitive to ModelViewer Primitive from which you can get the ModelViewer material and return it.